src/bundle/mutable.js

```diff
@@ -0,0 +1,144 @@
+// Adapted from SES/Caja
+// Copyright (C) 2011 Google Inc.
+// https://github.com/google/caja/blob/master/src/com/google/caja/ses/startSES.js
+// https://github.com/google/caja/blob/master/src/com/google/caja/ses/repairES5.js
+
+export default function makeRepairDataProperties() {
+  const {
+    defineProperties,
+    getOwnPropertyDescriptors,
+    hasOwnProperty,
+  } = Object;
+  const { ownKeys } = Reflect;
+
+  // Object.defineProperty is allowed to fail silently,
+  // wrap Object.defineProperties instead.
+  function defineProperty(obj, prop, desc) {
+    defineProperties(obj, { [prop]: desc });
+  }
+
+  /**
+   * For a special set of properties (defined below), it ensures that the
+   * effect of freezing does not suppress the ability to override these
+   * properties on derived objects by simple assignment.
+   *
+   * Because of lack of sufficient foresight at the time, ES5 unfortunately
+   * specified that a simple assignment to a non-existent property must fail if
+   * it would override a non-writable data property of the same name. (In
+   * retrospect, this was a mistake, but it is now too late and we must live
+   * with the consequences.) As a result, simply freezing an object to make it
+   * tamper proof has the unfortunate side effect of breaking previously correct
+   * code that is considered to have followed JS best practices, if this
+   * previous code used assignment to override.
+   *
+   * To work around this mistake, deepFreeze(), prior to freezing, replaces
+   * selected configurable own data properties with accessor properties which
+   * simulate what we should have specified -- that assignments to derived
+   * objects succeed if otherwise possible.
+   */
+  function beMutable(obj, prop, desc) {
+    if ('value' in desc && desc.configurable) {
+      const { value } = desc;
+
+      // eslint-disable-next-line no-inner-declarations
+      function getter() {
+        return value;
+      }
+
+      // Re-attach the data property on the object so
+      // it can be found by the deep-freeze traversal process.
+      getter.value = value;
```

Comment:
IIRC, this was one of the things that Salesforce changed, though I do not remember to what.

Reply:
its present in [forcedotcom/aura](https://github.com/forcedotcom/aura/blob/d77bb57ea56ddf769d2be961c3c97afecffc42ce/aura-resources/src/main/resources/aura/resources/lockerservice/aura-locker.js#L5367), don't have access to their latest work for comparison.

also present in node's version
https://github.com/nodejs/node/blob/33ae95c58da835086e6f9581715208c2d6ecfd70/lib/internal/freeze_intrinsics.js#L372

Reply:
`getter.value = value;` is fine.